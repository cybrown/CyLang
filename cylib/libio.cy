void extern putchar(int c);
void extern putbyte(byte c);
void extern putshort(short c);
void extern putint(int c);
void extern putlong(long c);
void extern putfloat(float f);
void extern putdouble(double d);
void extern puts(byte[] str);
void extern printf(byte[] str);

void print(byte c) {
	putbyte(c);
}

void print(short c) {
	putshort(c);
}

void print(int c) {
	putint(c);
}

void print(long c) {
	putlong(c);
}

void print(float f) {
	putfloat(f);
}

void print(double d) {
	putdouble(d);
}

void print(byte[] str) {
	printf(str);
}

void print(bool c) {
	print(c ? "true" : "false");
}
