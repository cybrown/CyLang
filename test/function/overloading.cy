import libio;

void take_int(int a) {
	putint(a);
	putchar(10);
}

void take_double(double f) {
	putdouble(f);
	putchar(10);
}

void take(int a) {
	puts("INT");
	take_int(a);
}

void take(double f) {
	puts("DOUBLE");
	take_double(f);
}

void take(byte[] str) {
	puts(str);
}

take(5);
take(3);
take(3.14);
take(42);
take(-45.32);
take(-34);
take(0);
take(0.0);
take("That's all folks!");
