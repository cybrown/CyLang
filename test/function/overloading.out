INT
5
INT
3
DOUBLE
3.140000
INT
42
DOUBLE
-45.320000
INT
-34
INT
0
DOUBLE
0.000000
That's all folks!
